packages/legend-query/src/components/QueryBuilderSetupPanel.tsx

```diff
@@ -79,6 +81,16 @@ const generateClassLabel = (
     val,
     queryBuilderState.graphManagerState.graph,
   );
+  const deprecatedProfile =
+    queryBuilderState.graphManagerState.graph.getProfile(
+      CORE_PURE_PATH.PROFILE_DOC,
+    );
+  let isDeprecatedClass = false;
+  if (
+    val.stereotypes.find((st) => st.ownerReference.value === deprecatedProfile)
```

Comment:
Although this technically works, its not fully correct. We should assert that the stereotype is the Deprecated one below. 

Another stereotype could be added to the list of stereotypes and would be caught in this condition. 
```
Profile meta::pure::profiles::doc
{
  stereotypes: [deprecated];
  tags: [doc, todo];
}
```